telebot/types.py

```diff
@@ -11429,3 +11429,350 @@ def de_json(cls, json_string):
+        OwnedGiftRegular
+        OwnedGiftUnique
+
+    Telegram documentation: https://core.telegram.org/bots/api#ownedgift
+    """
+    
+    
+    @classmethod
+    def de_json(cls, json_string):
+        if json_string is None: return None
+        obj = cls.check_json(json_string)
+        if obj["type"] == "regular":
+            return OwnedGiftRegular.de_json(obj)
+        elif obj["type"] == "unique":
+            return OwnedGiftUnique.de_json(obj)
+        
+class OwnedGiftRegular(OwnedGift):
+    """
+    This object describes a regular gift owned by a user or a chat.
+
+    Telegram documentation: https://core.telegram.org/bots/api#ownedgiftregular
+
+    :param type: Type of the gift, always “regular”
+    :type type: :obj:`str`
+
+    :param gift: Information about the regular gift
+    :type gift: :class:`Gift`
+
+    :param owned_gift_id: Optional. Unique identifier of the gift for the bot; for gifts received on behalf of business accounts only
+    :type owned_gift_id: :obj:`str`
+
+    :param sender_user: Optional. Sender of the gift if it is a known user
+    :type sender_user: :class:`User`
+
+    :param send_date: Date the gift was sent in Unix time
+    :type send_date: :obj:`int`
+    
+    :param text: Optional. Text of the message that was added to the gift
+    :type text: :obj:`str`
+
+    :param entities: Optional. Special entities that appear in the text
+    :type entities: :obj:`list` of :class:`MessageEntity`
+
+    :param is_private: Optional. True, if the sender and gift text are shown only to the gift receiver; otherwise, everyone will be able to see them
+    :type is_private: :obj:`bool`
+
+    :param is_saved: Optional. True, if the gift is displayed on the account's profile page; for gifts received on behalf of business accounts only
+    :type is_saved: :obj:`bool`
+
+    :param can_be_upgraded: Optional. True, if the gift can be upgraded to a unique gift; for gifts received on behalf of business accounts only
+    :type can_be_upgraded: :obj:`bool`
+
+    :param was_refunded: Optional. True, if the gift was refunded and isn't available anymore
+    :type was_refunded: :obj:`bool`
+
+    :param convert_star_count: Optional. Number of Telegram Stars that can be claimed by the receiver instead of the gift; omitted if the gift cannot be converted to Telegram Stars
+    :type convert_star_count: :obj:`int`
+
+    :param prepaid_upgrade_star_count: Optional. Number of Telegram Stars that were paid by the sender for the ability to upgrade the gift
+    :type prepaid_upgrade_star_count: :obj:`int`
+
+    :return: Instance of the class
+    :rtype: :class:`OwnedGiftRegular`
+    """
+    def __init__(self, type, gift, owned_gift_id=None, sender_user=None, send_date=None, text=None, entities=None,
+                    is_private=None, is_saved=None, can_be_upgraded=None, was_refunded=None, convert_star_count=None,
+                    prepaid_upgrade_star_count=None, **kwargs):
+        self.type: str = type
```

Comment:
Propose to move "type" declaration to parent class.

Also may be "gift" with Union... May be not.

Reply:
What do you mean with Union?

Reply:
put "gift" to the parent class with the type
gift: Union[Gift, UniqueGift]...